pkg/provider/provider_helpers.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mitchellh/go-homedir"
+	"github.com/youmark/pkcs8"
+	"golang.org/x/crypto/ssh"
+)
+
+func mergeSchemas(schemaCollections ...map[string]*schema.Resource) map[string]*schema.Resource {
```

Comment:
It would be nice to unit test the methods from this file (I know that there are other helpers throughout the project, which also need testing, but I would like to move towards testing such things). It does not have to be done in this PR, it does not necessarily have to be the next one - it can be added as a part of SNOW-933289 (please create a subtask there if you choose to postpone it).

Reply:
i would sooner refactor the code to not use this function at all than to write a test for it, because this function is not very useful anyways. but I get what you are saying